client/src/components/chat/chatTab.js

```diff
@@ -51,7 +54,37 @@ export default class chatTab extends Component {
             if (data.chatId !== this.props.chatId){
                 return;
             }
-            self.setState( { messages : self.state.messages.concat([data]) } );
+
+            this.setState( {
+                showIsTyping: '',
+                messages : self.state.messages.concat([data])
+            })
+        })
+
+        this.socket.on(`server:userConnected`, data => {
+
+            if (!this.ignoredOwnConnectMessage){
```

Comment:
Alternatively, is it possible to emit the isTyping event with an id, and the server won't send this event back to where it came from?